### DISCUSSION TITLE
How to get count of nested data ?

### DISCUSSION THREAD

**Discussion by dukesx** [supabase/supabase#8691] [Questions]
I have a schema where i fetch articles related to tags followed by a user (like on dev.to). I am using Limit() function to limit the results returned for infinite loading. But it seems that the count which is returned , is for amount of tags followed, not the amount of articles. This is a complex situation. When i used Hasura, this was easy, i don't know how but it used to return the cumulative amount of articles returned (which is nested data of-course). This is my structure query on supabase-js :
```gql
            tags  (
                  title,
                  articles    (
                    id,
                    title,
                    description,
                    cover
         )
    )
```

Authors is related to tags through author_followed_tags table which has two PKs, author id and tag id. Articles are related to tags through articles_tags table which also has two PKs , article_id and tag_id.  I am unable to paginate this right now. 

Authors > author_followed_tags > tags > articles_tags > articles > {data}

Any advice on what can be done ? I need this tag based follow system for customisable articles feed for registered users.

**Comment by steve-chavez** [comment]
Try using `!inner`
```
 articles!inner(
  id,
  ...
```
That should modify the total count based on an embedded table.

**Reply by dukesx** [reply]
Great, Thank you ! one more thing, lets say, now i fetch articles directly based on an array column called tags and i use ContainedBy to filter them. In this situation , what can be done to get total number of articles ? i mean i get count of returned rows, thats fine but for pagination , i need total number of rows that match the query ? 

```tsx
  const {
            error,
            data: data2,
            count: count1,
          } = await supabaseClient
            .from("articles")
            .select(
              `
                  id,
                  title,
                  description,
                  cover,
                  authors (
                    dp,
                    firstName,
                    lastName
                  ),
                  co_authors_articles (
                    authors (
                      dp,
                      firstName,
                      lastName
                    )
                  )
              `,
              { count: "exact" }
            )
            .containedBy("article_tags", authorFollowedTags[0].followed_tags)
            .limit(5)
            .range(0, 5);

```

**Reply by dukesx** [reply]
@steve-chavez i tried using inner but it doesnt modify the count. It remains the same.

**Reply by steve-chavez** [reply]
Can you share a snippet of what you tried?

**Reply by steve-chavez** [reply]
`!inner` should modify the top count if all its embedded resources also use `!inner`.
